ui/app/src/App.tsx

```diff
@@ -61,8 +67,50 @@ function App() {
     return Math.random().toString(36).substring(7);
   };
 
-  // Get stream from CreateStream to pass it as props
+  const publicClient = usePublicClient();
+  const walletClient = useWalletClient()?.data;
+  const { address } = useAccount();
+
   const [stream, setStream] = useState<Stream | null>(null);
+  const [SM, setSM] = useState<StreamManager | null>(null);
+  const [createdStreams, setCreatedStreams] = useState<Stream[]>([]);
```

Comment:
would have it manage all streams from the contract, and then add a method to the JS SDK to fetch all past streams to start with

when you fetch the `StreamManager` I would run through all past events to catch up to present, and then run the `.onStreamCreated` watcher to listen to new streams created